realm/src/main/java/io/realm/Realm.java

```diff
@@ -656,6 +687,15 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
                 throw new IllegalArgumentException("Two configurations with different schemas are trying to open " +
                         "the same Realm file. Their schema must be the same: " + newConfiguration.getPath());
             }
+
+            // Check if the durability is the same
+            SharedGroup.Durability cachedDurability = cachedConfiguration.getDurability();
+            SharedGroup.Durability newDurability = newConfiguration.getDurability();
+            if (cachedDurability.value != newDurability.value) {
+                throw new IllegalArgumentException("Configurations cannot have different durabilities for the " +
+                        "same realm filename/identifier: " + newConfiguration.getRealmFileName() + " " +
```

Comment:
We should probably at least check why they decided to go with identifier instead of name. Maybe it is to avoid confusion with the path, but as an in-memory Realm still might write data to disc we cannot really separate the two IMO.